### ISSUE TITLE
Switching Auth flow to USER_PASSWORD_AUTH is not working

### ISSUE THREAD

**Issue by kapi-ai** [aws-amplify/aws-sdk-android#2349]
**Describe the bug**
I am using the Amplify library in an App written in Kotlin. While experimenting with the "authenticationFlowType" setting in my amplifyconfiguration.json file, I noticed that it is impossible to switch to USER_PASSWORD_AUTH, even, if you set it in the amplifyconfiguration.

The problem here lies in  the Sign-In functionality of the AWSMobileClient.java of the aws-android-sdk-mobile-client. The function _sign_in internally overrides the function getAuthenticationDetails() in line 1207 (in tag 2.22.1). 
In this override, new AuthenticationDetails are created depending on the awsConfiguration. New AuthenticationDetails are created and the default constructor always uses "CognitoServiceConstants.CHLG_TYPE_USER_PASSWORD_VERIFIER" as authenticationType so that is what we end up with in the AuthenticationDetails.
Now, when we want to initiate the user authentication in _initiateUserAuthentication() in line 1178 of CognitoUser.java we check using some if/elseif blocks the authenticationType. Since it is by default always  CHLG_TYPE_USER_PASSWORD_VERIFIER we can never go to the function startWithUserPasswordAuth() but always go to startWithUserSrpAuth().

The root of the problem seems to be hardcoding the value for authenticationType when constructing new AuthenticationDetails and then not checking them after creation. This should be checked or changed. 

**To Reproduce**
A code sample or steps:
amplifyconfig.json
```
{
  "auth": {
    "plugins": {
      "awsCognitoAuthPlugin": {
        "IdentityManager": {
          "Default": {}
        },
        "CredentialsProvider": {
          "CognitoIdentity": {
            "Default": {
              "PoolId": "XXX",
              "Region": "XXX"
            }
          }
        },
        "CognitoUserPool": {
          "Default": {
            "PoolId": "XXX",
            "AppClientId": "XXX",
            "Region": "XXX"
          }
        },
        "Auth": {
          "Default": {
            "authenticationFlowType": "USER_PASSWORD_AUTH"
          }
        }
      }
    }
  }
}
```
Login part in AuthActivity.kt:
```
     Amplify.Auth.signIn(
            username,
            password,
            { result ->
                runOnUiThread(Runnable {
                    Toast
                        .makeText(
                            this@AuthenticatorActivity,
                            "Logged in",
                            Toast.LENGTH_LONG
                        )
                        .show()
                })
            },
            { error ->
                Log.e("AuthQuickstart", error.toString())
                Log.e("AuthQuickstart", error.message.toString())
                Log.e("AuthQuickstart", error.stackTraceToString())
            }
        )
    }
```
See 

**Which AWS service(s) are affected?**

**Expected behavior**
Have User-Password authentication instead of SRP authentication

**Screenshots**

**Environment Information (please complete the following information):**
 - AWS Android SDK Version: 2.22.1
 - Device: Simulator
 - Android Version: 8.0.0
 - Specific to simulators: -

**Additional context**
I am not an expert in Amplify development that is why it is unclear for me if I have done something wrong or if the code itself is buggy. It is such an obvious problem, yet there exists no bug report for it, so I am confident that the error is actually on my side.
However, I am kind of stuck at this point, so I though having this bug report might help.

**Comment by bjornshala** [comment]
@kapi-ai Hi, did you find a solution for this issue? I'm having the same problem

**Comment by Abhishek0786** [comment]
you changes has been reverted in 2.22.5 version of aws-sdk-android. 
is anybody help me how to solve this problem in latest version 2.22.6 of aws-sdk-android?

**Comment by rjuliano** [comment]
This was fixed in version 2.28.0
